src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPv4InterfaceProperties.cs

```diff
@@ -34,9 +34,28 @@ public LinuxIPv4InterfaceProperties(LinuxNetworkInterface linuxNetworkInterface)
 
         private bool GetIsForwardingEnabled()
         {
-            // /proc/sys/net/ipv4/conf/<name>/forwarding
-            string path = Path.Combine(NetworkFiles.Ipv4ConfigFolder, _linuxNetworkInterface.Name, NetworkFiles.ForwardingFileName);
-            return StringParsingHelpers.ParseRawIntFile(path) == 1;
+            string[] paths = new string[]
+            {
+                // /proc/sys/net/ipv4/conf/<name>/forwarding
+                Path.Join(NetworkFiles.Ipv4ConfigFolder, _linuxNetworkInterface.Name, NetworkFiles.ForwardingFileName),
+                // Fall back to global forwarding config /proc/sys/net/ipv4/ip_forward
+                NetworkFiles.Ipv4GlobalForwardingFile
+            };
```

Comment:
Eventually if/when we support stackalloc of references, this could be changed to `Span<string> span = stackalloc string[] { ... }` to avoid the array allocation.  In the meantime, if we wanted to avoid it, you could just store the two strings into their own locals, e.g.
```C#
string path1 = ...;
string path2 = ...;
```
and then change the loop to be something like:
```C#
for (int i = 0; i < 2; i++)
{
    string path = i == 0 ? path1 : path2;
    ...
}
```
That said, there's a lot of allocation happening anyway in ParseRawIntFile, including everything associated with opening and parsing the file, so the extra string[] is unlikely to matter at all.

So, it's fine as is, but wanted to call it out.